threatconnect.js

```diff
@@ -533,28 +533,72 @@ class ThreatConnect {
     });
   }
 
-  _enrichResult(indicatorType, indicatorValue, result) {
-    result.meta = {
-      indicatorType: indicatorType,
-      indicatorValue: indicatorValue
-    };
+  getDnsInformation(indicatorTypePlural, indicatorValue, owner, cb) {
+    let self = this;
+    let indicatorTypeSingular;
 
-    if (typeof result.rating === 'undefined') {
-      result.rating = 0;
+    if (indicatorTypePlural === 'addresses' || indicatorTypePlural === 'hosts') {
+      indicatorTypeSingular = 'dnsResolution';
     }
 
-    result.ratingHuman = this._getRatingHuman(result.rating);
-
-    if (typeof result.confidence === 'undefined') {
-      result.confidence = 0;
+    if (typeof indicatorTypeSingular === 'undefined') {
+      return cb({
+        detail: `The provided indicator type '${indicatorTypePlural}' is invalid`
+      });
     }
-    result.confidenceHuman = this._getConfidenceHuman(result.confidence);
 
-    if (!Array.isArray(result.tag)) {
-      result.tag = [];
-    }
+    this._getDnsInformation(indicatorTypePlural, indicatorValue, owner, function (err, response, body) {
+      self._formatResponse(err, response, body, function (err, data) {
+        if (err || !data) return cb(err, data);
+
+        let result = data['dnsResolution'];
+
+        if (Object.keys(data.dnsResolution).length > 0) {
+          result = self._enrichResult(indicatorTypePlural, indicatorValue, result);
```

Comment:
Whenever I search any address that exists in TC, including `37.120.193.123`, `185.181.8.49`, `198.71.232.3`, `75.119.204.33` etc, I get a timeout error and this error when `onDetails` runs
```
{
    "debug": {
        "message": "Cannot convert undefined or null to object",
        "stack": "TypeError: Cannot convert undefined or null to object\n    at Function.keys (<anonymous>)\n    at /app/polarity-server/integrations/threatconnect/threatconnect.js:557:20\n    at ThreatConnect._formatResponse (/app/polarity-server/integrations/threatconnect/threatconnect.js:881:7)\n    at Request._callback (/app/polarity-server/integrations/threatconnect/threatconnect.js:552:12)\n    at Request.self.callback (/app/polarity-server/integrations/threatconnect/node_modules/request/request.js:185:22)\n    at Request.emit (events.js:314:20)\n    at Request.<anonymous> (/app/polarity-server/integrations/threatconnect/node_modules/request/request.js:1154:10)\n    at Request.emit (events.js:314:20)\n    at IncomingMessage.<anonymous> (/app/polarity-server/integrations/threatconnect/node_modules/request/request.js:1076:12)\n    at Object.onceWrapper (events.js:420:28)"
    },
    "error-signature": "92262f8c2866219d6d33ae5c5a2787af"
}
```

<img width="600" alt="Screen Shot 2021-09-01 at 5 43 58 PM" src="https://user-images.githubusercontent.com/58269952/131759463-90653409-2ec7-4a2c-9fad-70e0d66bd863.png">